overlord/configstate/config/transaction.go

```diff
@@ -46,6 +46,23 @@ type Transaction struct {
 	changes  map[string]map[string]interface{}
 }
 
+// RunTransaction holds a transaction with a task that is in charge of
+// appliying a change to the configuration. It is used in the context of
+// configcore.
+type RunTransaction struct {
```

Comment:
didn't we discuss that this change should go to configcore itself? it should also be a separate PR?